### PR TITLE
Update funannotate to 1.8.9

### PR DESCRIPTION
**Pull request by BiocondaBot**
Update [`funannotate`](https://bioconda.github.io/recipes/funannotate/README.html): **1.8.7** &rarr; **1.8.9**

[install with bioconda](http://bioconda.github.io/recipes/funannotate/README.html) [Conda](https://anaconda.org/bioconda/funannotate/files)

Info | Link
-----|-----
Recipe | [`recipes/funannotate`](https://github.com//bioconda/bioconda-recipes/tree/bump/funannotate/recipes/funannotate) (click to view/edit other files)
Releases |[https://pypi.org/pypi/funannotate/json](https://pypi.org/pypi/funannotate/json)
Recipe Maintainer(s) | @nextgenusfs

***

This pull request was automatically generated (see [docs](https://bioconda.github.io/contributor/updating.html)).

### CODE REVIEW

**Comment by nextgenusfs** [comment]
@dpryan79 looking for some advice here -- what do I do when the build times out? It seems like it perhaps cannot solve? It is a complicated environment and locally I switched to using mamba as its solves much quicker, ie this is building fine via GitHub actions: https://github.com/nextgenusfs/funannotate/blob/master/Dockerfile. This has solved/built successfully in previous releases -- really no change to the environment from the last release just some updated python code.

**Comment by dpryan79** [comment]
@nextgenusfs If you're able to solve the environment with mamba then start restricting the dependency versions based on that so the conda solver can resolve the environment.

**Review by nextgenusfs — APPROVED** [review]

File: recipes/funannotate/meta.yaml [hunk 1]
```diff
@@ -17,64 +17,66 @@ build:
...
-    - proteinortho >=6.0.9
-    - salmon >=0.9
-    - perl
-    - perl-bioperl >1.7
-    - perl-dbd-mysql
-    - perl-clone
+    - glimmerhmm  ==3.0.4
+    - gmap  ==2017.11.15
+    - bamtools  ==2.5.1 # [linux]
+    - augustus  ==3.3.3  # [linux]
```
Comment (abretaud) [inline recipes/funannotate/meta.yaml:57]: From my local tests it feels like the slowness comes from this augustus version restriction, any reason not to update to 3.4.0?

Comment (nextgenusfs) [reply recipes/funannotate/meta.yaml:57]: Yeah it's not compatible with the internal BUSCO method. What I can't figure out is that it seems to build if I specified the host python as 3.8, but I think that would limit it to work only if users were running 3.8? It does that not matter?

Comment (abretaud) [reply recipes/funannotate/meta.yaml:57]: I've made https://github.com/bioconda/bioconda-recipes/pull/30152 to try to rebuild the augustus 3.3.3, maybe that will help :crossed_fingers:

**Comment by abretaud** [comment]
Ah, green now! Does it look good to you @nextgenusfs?
@bioconda/core I feel bad about pining augustus to a specific build, but pining only to 3.3.3 makes the build unfinishable, what do you think?

**Comment by abretaud** [comment]
@BiocondaBot please merge

**Comment by nextgenusfs** [comment]
Thanks @abretaud foe the help. I lost my patience after a week or so. What I was concerned about with pinning the host python to some versions, does that mean if a user has miniconda with python 3.9 that this won't install for them? Or is the host python pin only used for building?

**Comment by abretaud** [comment]
No problem, I'm happy to help :)

The constraint means that conda won't be able to create an environment with both funannotate and python 3.9, it will switch to 3.8 by default, unless you get a conflict with some other package requiring python 3.9

Looking at the logs, the build failed with python 3.9 only because of https://github.com/nextgenusfs/funannotate/blob/master/setup.py#L20. If there's no specific incompatibility with py3.9, maybe you'd just need to update this line to get it built for it too

**Comment by nextgenusfs** [comment]
The 3.9 incompatibility is related to python multiprocessing changes, I need to update the code for it to work for 3.9. Which I think as of last week I finally figured out why it was failing in >3.8 has to do with the default method that multiprocessing module initializes.

**Comment by abretaud** [comment]
Ok, then we can leave like this for now, and change it later when you have a new release supporting py3.9

**Pull request merged**